internal/controller/drplacementcontrol_controller.go

```diff
@@ -565,40 +565,36 @@ func GetDRClusters(ctx context.Context, client client.Client, drPolicy *rmn.DRPo
 }
 
 // updateObjectMetadata updates drpc labels, annotations and finalizer, and also updates placementObj finalizer
-func (r DRPlacementControlReconciler) updateObjectMetadata(ctx context.Context,
-	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
-) error {
-	var update bool
+func (r DRPlacementControlReconciler) updateObjectMetadata(
+	drpc *rmn.DRPlacementControl, placementObj client.Object,
+) (drpcModified bool, placementModified bool, err error) {
```

Comment:
Why we change both objects in this function? It will be better split this to 2 functions , each modifying one object. This avoids the error prone function return value (bool, bool, error). It is very easy to call the function in the wrong way:

    drpcModified, placementModified, err := r.updateObjectMetadata()

    placementModified, drpcModified, err := r.updateObjectMetadata()

Which call it the correct one? The type system does not help with such code.